### ISSUE TITLE
[BUG] Suspend under sway causes system to freeze

### ISSUE THREAD

**Issue by seoushi** [fedora-silverblue/issue-tracker#319]
**Description**
I layered on sway and if I run "systemctrl suspend the monitors will turn off and the system stays on and am forced to hard power off the system. Suspend under gnome however works fine.

**To Reproduce**
1. install the sway package and reboot (rpm-ostree install sway)
2. open a term (super+enter) is the default to spawn on in sway
3. run "systemctl suspend"

**Expected behavior**
System goes to sleep like it does in gnome

**OS version:**
```
● fedora:fedora/36/x86_64/silverblue
                   Version: 36.20220724.0 (2022-07-24T00:46:41Z)
                BaseCommit: 9c188835f185b9ef75698aff61e046bfc57e2547c38cdcff5e35dd4f1729bdb9
              GPGSignature: Valid signature by 53DED2CB922D8B8D9E63FD18999F7CBF38AB71F4
           LayeredPackages: digikam distrobox emacs flameshot ifuse libheif
                            libimobiledevice-utils mako neovim qt-heif-image-plugin sddm
                            sway waybar
             LocalPackages: rpmfusion-free-release-36-1.noarch
                            rpmfusion-nonfree-release-36-1.noarch
```

**Additional context**
This looks to be an issue with the amdgpu driver. Here is a
[systemlog](https://github.com/fedora-silverblue/issue-tracker/files/9176334/systemlog.txt) for that boot.
the more interesting part is:
```
Jul 24 10:19:08 foresti kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring sdma0 timeout, signaled seq=1675, emitted seq=1677
Jul 24 10:19:08 foresti kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* Process information: process  pid 0 thread  pid 0
Jul 24 10:19:08 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: GPU reset begin!
Jul 24 10:19:09 foresti kernel: [drm:dc_dmub_srv_wait_idle [amdgpu]] *ERROR* Error waiting for DMUB idle: status=3
Jul 24 10:19:09 foresti kernel: amdgpu 0000:0c:00.0: [drm:amdgpu_ring_test_helper [amdgpu]] *ERROR* ring kiq_2.1.0 test failed (-110)
Jul 24 10:19:09 foresti kernel: [drm:gfx_v10_0_hw_fini [amdgpu]] *ERROR* KGQ disable failed
Jul 24 10:19:09 foresti kernel: amdgpu 0000:0c:00.0: [drm:amdgpu_ring_test_helper [amdgpu]] *ERROR* ring kiq_2.1.0 test failed (-110)
Jul 24 10:19:09 foresti kernel: [drm:gfx_v10_0_hw_fini [amdgpu]] *ERROR* KCQ disable failed
Jul 24 10:19:12 foresti chronyd[1341]: Selected source 119.28.183.184 (2.fedora.pool.ntp.org)
Jul 24 10:19:14 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: SMU: I'm not done with your previous command: SMN_C2PMSG_66:0x00000036 SMN_C2PMSG_82:0x00000000
Jul 24 10:19:14 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: Failed to disable smu features.
Jul 24 10:19:14 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: Fail to disable dpm features!
Jul 24 10:19:14 foresti kernel: [drm:amdgpu_device_ip_suspend_phase2 [amdgpu]] *ERROR* suspend of IP block <smu> failed -62
Jul 24 10:19:14 foresti kernel: [drm] free PSP TMR buffer
Jul 24 10:19:15 foresti kernel: [drm] psp gfx command DESTROY_TMR(0x7) failed and response status is (0x80000306)
Jul 24 10:19:15 foresti kernel: CPU: 0 PID: 2642 Comm: kworker/u64:49 Not tainted 5.18.13-200.fc36.x86_64 #1
Jul 24 10:19:15 foresti kernel: Hardware name: Gigabyte Technology Co., Ltd. X570 AORUS ELITE WIFI/X570 AORUS ELITE WIFI, BIOS F37c 05/12/2022
Jul 24 10:19:15 foresti kernel: Workqueue: amdgpu-reset-dev drm_sched_job_timedout [gpu_sched]
Jul 24 10:19:15 foresti kernel: Call Trace:
Jul 24 10:19:15 foresti kernel:  <TASK>
Jul 24 10:19:15 foresti kernel:  dump_stack_lvl+0x44/0x5c
Jul 24 10:19:15 foresti kernel:  amdgpu_do_asic_reset+0x26/0x46f [amdgpu]
Jul 24 10:19:15 foresti kernel:  amdgpu_device_gpu_recover_imp.cold+0x601/0x8af [amdgpu]
Jul 24 10:19:15 foresti kernel:  amdgpu_job_timedout+0x156/0x190 [amdgpu]
Jul 24 10:19:15 foresti kernel:  ? resched_curr+0x20/0xc0
Jul 24 10:19:15 foresti kernel:  ? _raw_spin_unlock_irqrestore+0x23/0x40
Jul 24 10:19:15 foresti kernel:  drm_sched_job_timedout+0x76/0x110 [gpu_sched]
Jul 24 10:19:15 foresti kernel:  process_one_work+0x1c7/0x380
Jul 24 10:19:15 foresti kernel:  worker_thread+0x4d/0x380
Jul 24 10:19:15 foresti kernel:  ? process_one_work+0x380/0x380
Jul 24 10:19:15 foresti kernel:  kthread+0xe9/0x110
Jul 24 10:19:15 foresti kernel:  ? kthread_complete_and_exit+0x20/0x20
Jul 24 10:19:15 foresti kernel:  ret_from_fork+0x22/0x30
Jul 24 10:19:15 foresti kernel:  </TASK>
Jul 24 10:19:15 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: MODE1 reset
Jul 24 10:19:15 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: GPU mode1 reset
Jul 24 10:19:15 foresti kernel: amdgpu 0000:0c:00.0: amdgpu: GPU smu mode1 reset
```

**Comment by travier** [comment]
As this is not the default desktop environment for Silverblue, you should probably report that to Sway developers as we're unlikely to work on that. It also looks like you have a kernel stack trace so it's likely a kernel issue and should be reported in either the Fedora Bugzilla or directly upstream in https://gitlab.freedesktop.org/drm/amd/-/issues.

**Comment by seoushi** [comment]
Thanks. I looked into this further and it doesn't seem sway related either. If I disable the login manager and login from a shell and run ```systemctl suspend``` it freezes up as well. I was able to find a work around incase others have the same issue. By default fedora uses the acpi_cpfreq and I notice that my gentoo system which works uses amd_pstate. I switched over to the amd_pstate driver and suspend works.
```
rpm-ostree kargs --append=initcall_blacklist=acpi_cpufreq_init --append=amd_pstate.shared_mem=1
```
